r/tests/testthat/test-RecordBatch.R

```diff
@@ -571,7 +571,7 @@ test_that("RecordBatchReader to C-interface", {
 
   # export the RecordBatchReader via the C-interface
   stream_ptr <- allocate_arrow_array_stream()
-  on.exit(delete_arrow_array_stream(stream_ptr))
+  on.exit(delete_arrow_array_stream(stream_ptr), add = TRUE)
```

Comment:
This is probably not worth changing, but I think `add = TRUE` is not required here, since this is the first `on.exit()` in the function, correct?

Reply:
yes, that's true — I put it here as a reminder that if we move it around / add another one we need to make sure to have it, though it is not strictly necessary.

Reply:
FWIW the `add = TRUE` argument does not exist in all versions of R that we support today (I can't remember whether it was added in 3.4 or 3.5).

Reply:
Aaah good catch, I've moved all of these to calling after we don't need the pointer anymore (with a comment). That also calls out _slightly more clearly_ that pointer cleanup is *not* part of these functions which is not bad